packages/cozy-client/src/authentication/mobile.js

```diff
@@ -6,9 +6,21 @@ import {
   isIOSApp
 } from 'cozy-device-helper'
 
+/**
+ * @typedef {object} WindowInterface
+ * @property {object} SafariViewController
+ * @property {object} cordova
+ * @property {Function} handleOpenURL
+ */
+
+/**
+ * @type {WindowInterface}
+ */
+const windowInterface = window
```

Comment:
Did you do this to not have to deal with all the props of `window`?

Reply:
This file uses global properties like SafariViewController, cordova and handleOpenURL. I did not manage to get the typescript compiler to accept that those properties exist on the window object, so I created this interface. This lets me ignore the typescript error only once later on.

```
/**
 * @type {WindowInterface}
 */
// @ts-ignore
const windowInterface = window
```

Reply:
I extracted the CordovaWindow type to the "types" file.

I am not completely happy with the necessary @ts-ignore but it works.